core/kt-osrd-utils/src/test/kotlin/fr/sncf/osrd/utils/TestDistanceRangeMap.kt

```diff
@@ -91,4 +92,45 @@ class TestDistanceRangeMap {
             DistanceRangeMap.RangeMapEntry(Distance(0), Distance(500), 42),
         ), rangeMap.asList())
     }
+
+    @Test
+    fun testTruncate() {
+        val rangeMap = DistanceRangeMapImpl<Int>()
+        rangeMap.put(Distance(0), Distance(100), 41)
+        rangeMap.put(Distance(200), Distance(300), 42)
+        rangeMap.truncate(Distance(250), Distance(260))
+        assertEquals(listOf(
+            DistanceRangeMap.RangeMapEntry(Distance(250), Distance(260), 42),
+        ), rangeMap.asList())
+    }
+
+    @Test
+    fun testTruncateAll() {
+        val rangeMap = DistanceRangeMapImpl<Int>()
+        rangeMap.put(Distance(0), Distance(100), 41)
+        rangeMap.put(Distance(200), Distance(300), 42)
+        rangeMap.truncate(Distance(0), Distance(0))
+        assertEquals(listOf(), rangeMap.asList())
+    }
+
+    @Test
+    fun testTruncateToEmptyRange() {
+        val rangeMap = DistanceRangeMapImpl<Int>()
```

Comment:
there's a standard kotlin convention of having a builder function as so: `distanceRangeMapOf<Int>()`
(to avoid the nasty Impl constructor call)